platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,714 @@
+        LibraryLoader.load(LIB_HEDERA_BLS);
+    }
+}
+```
+The method will fail with a runtime exception if no description matches the current system architecture.
+The result of this invocation will be that the library will be packaged from the contained jar in the expected structure into a temporal folder and loaded into memory.   
+
+## Test Plan
+Since cryptographic code is often difficult to test due to its complexity and the lack of a test oracle,
+we should design our test cases based on the cryptographic properties that these implementations should satisfy.
+
+
+**Properties**:
+
+Protection of private keys. Our scheme protects the secrecy of the group’s and members’ private keys; otherwise, private keys can generate only one group signature.
+
+Unforgibility of group signature. Our scheme ensures that (a) any t or more than t members can work
+together to generate a valid group signature, and (b)fewer than t members cannot generate a valid group signature.
+
+Fixed length of threshold signature. Our scheme ensures that the size of a threshold signature is fixed (i.e., not depending on the number of signers).
+
+Efficiency of verification. Verifying a group signature is based on the group’s public key.
+
+* Generation of Private and Public Key Pairs: Tests the generation of key pairs for correctness by having the IUT produce 10 key
+  pairs. The private key provided is used to compute the public key, Q’. The calculated value Q’ is then compared to the supplied public key, Q.
+* Public Key Validation: Generate 12 key pairs for each supported curve, modify some public keys to introduce errors, and determine whether it can detect them.
+* Signature Generation: Ten preconfigured known messages per curve supported are supplied. Generate a signature for each message. The message, public key, and signature components are returned.
+  The signatures are compared against previous known signatures.
+* Signature Verification: For each curve supported, 15 pseudorandom messages, a public key, Q, and a signature component (r, s) are supplied if some of the provided values are modified so that signature verification should fail.
+
+Some ideas:
+* [Cryptographic-Algorithm-Validation-Program](https://csrc.nist.gov/CSRC/media//Projects/Cryptographic-Algorithm-Validation-Program/documents/dss/ECDSAVS.pdf):
+* https://csrc.nist.gov/CSRC/media/Events/lightweight-cryptography-workshop-2019/documents/papers/systematic-testing-of-lightweight-crypto-lwc2019.pdf
+
+The proposal is to move forward with the implementation while creating a work group to discuss and collect ideas for testing and validating these features.
+**Questions**:
+* What other properties are not mentioned here?
+
+### Unit Tests
+One of the components that can be unit-tested is native support. SPI loading and failing. JNI wrapping.
+Limitations: native-support-library tests must be executed in different environments to provide comprehensive code validation.
+
+### Performance Testing
```

Comment:
We should know that variance (min, max, average, median) time it takes to do the following: 

1. Genesis for a 4, 10, 20, 40, and 100 node network. 
2. Rekeying for a 4, 10, 20, 40, and 100 node network. 
3. Shares Amounts distributed randomly:  200, 500, 1000, 2000, 5000

we need to test each configuration 100 times to get confident mins, maxes, and a good average and median

Reply:
Added the notes, thanks